### PR TITLE
Added hyperbolic trigonometric functions.

### PR DESCRIPTION
**Pull request by blas-ko**
Added functions `sinh`, `cosh` , `tanh` to `Taylor1.jl` alongside with some tests. 

*NOTE:* For proving that  ` -sinh(t)^2 + cosh(t)^2 == 1`  a random number `rand()` is evaluated on the tests. Is this legit enough?

### CODE REVIEW

**Comment by lbenet** [comment]
It looks good to me. Yet, due to some issue with travis, some tests are not passing. I have first to update travis, and then proceed to merge.

File: src/Taylor1.jl [hunk 1]
```diff
@@ -981,6 +981,133 @@ function atanHomogCoef{T<:Number}(kcoef::Int, ac::Array{T,1}, rc::Array{T,1}, co
     coefhomog
 end
 
+### HYPERBOLIC TRIGONOMETRIC FUNCTIONS ###
```
Comment (lbenet) [inline src/Taylor1.jl]: "TRIGONOMETRIC" should be erased; the functions are simply called hyperbolic functions.

File: test/runtests.jl [hunk 2]
```diff
@@ -102,6 +102,12 @@ facts("Tests for Taylor1 expansions") do
     @fact asin(t) + acos(t) == pi/2 --> true
     @fact derivative(acos(t)) == - 1/sqrt(1-t^2) --> true
 
+    @fact - sinh(t) + cosh(t) == exp(-t) --> true
+    @fact  sinh(t) + cosh(t) == exp(t) --> true
+    @fact evaluate(- sinh(t)^2 + cosh(t)^2 , rand()) == 1 --> true
+    @fact tanh(t + 0im) == -1im * tan(t*1im) --> true
+    @fact  evaluate(tanh(t/2),1.5) == evaluate(sinh(t) / (cosh(t) + 1),1.5) --> true
+
```
Comment (lbenet) [inline test/runtests.jl]: It's worth to add tests that involve identities with complex numbers and other functions. For instance,
```julia
@fact cosh(t) == real(cos(im*t)) --> true
@fact sinh(t) == imag(sin(im*t)) --> true
```

File: test/runtests.jl [hunk 3]
```diff
@@ -102,6 +102,12 @@ facts("Tests for Taylor1 expansions") do
     @fact asin(t) + acos(t) == pi/2 --> true
     @fact derivative(acos(t)) == - 1/sqrt(1-t^2) --> true
 
+    @fact - sinh(t) + cosh(t) == exp(-t) --> true
+    @fact  sinh(t) + cosh(t) == exp(t) --> true
+    @fact evaluate(- sinh(t)^2 + cosh(t)^2 , rand()) == 1 --> true
```
Comment (lbenet) [inline test/runtests.jl]: I understand that numerical errors propagate, and make that the actual identity is not identically fulfilled, which motivates the evaluation. Checking this in more detail, it seems that only beyond order 8, these problems appear. For the time being, we can leave this as it is.

**Comment by blas-ko** [comment]
Commented changes done!

File: test/runtests.jl [hunk 4]
```diff
@@ -102,6 +102,14 @@ facts("Tests for Taylor1 expansions") do
     @fact asin(t) + acos(t) == pi/2 --> true
     @fact derivative(acos(t)) == - 1/sqrt(1-t^2) --> true
 
+    @fact - sinh(t) + cosh(t) == exp(-t) --> true
+    @fact  sinh(t) + cosh(t) == exp(t) --> true
+    @fact evaluate(- sinh(t)^2 + cosh(t)^2 , rand()) == 1 --> true #works in an eps() difference, but ideally it should return a 1 + 𝒪(t¹⁶) Taylor.
```
Comment (lbenet) [inline test/runtests.jl]: The comment is appreciated!

**Comment by lbenet** [comment]
Great! Thanks for the contribution! Tests are passing, so I'm merging!

**Pull request merged**